testing/python/collect.py

```diff
@@ -1018,18 +1018,18 @@ class TestReportInfo:
     def test_itemreport_reportinfo(self, testdir):
         testdir.makeconftest(
             """
-            import pytest
+            import pytest, py
             class MyFunction(pytest.Function):
                 def reportinfo(self):
-                    return "ABCDE", 42, "custom"
+                    return py.path.local("foo"), 42, "custom"
```

Comment:
That's kind of an API change - I've noticed this myself that the tests seem to be the only things using `str`.
Ok with me in general, just wanted to mention this, i.e. some plugins might return strings there (but likely fail then afterwards, depending on where it is used I guess?)
Maybe this could/should have a test?

Reply:
The `_bestrelpathcache` backing this only works with `py.path.local`, that it returned something for `str` was accidental and incorrect. So I think the assert just codifies the current behavior.

Only this test trigerred the assert, and it's just a mock, so I think it should be fine.